crash with azurerm_linux_virtual_machine_scale_set

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

terraform 1.0.4
azurerm v2.72.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_linux_virtual_machine_scale_set

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
locals = {
  all_nginx_instances = {
    "dev" = {
      [...]
      max_batch_instance_percent              = 100
      max_unhealthy_instance_percent          = 100
      max_unhealthy_upgraded_instance_percent = 100
      pause_time_between_batches              = "PT120S"
    }
}

resource "azurerm_linux_virtual_machine_scale_set" "scalesetnginx" {
  [...]
  rolling_upgrade_policy {
    max_batch_instance_percent              = each.value.max_batch_instance_percent
    max_unhealthy_instance_percent          = each.value.max_unhealthy_instance_percent
    max_unhealthy_upgraded_instance_percent = each.value.max_unhealthy_upgraded_instance_percent
    pause_time_between_batches              = each.value.pause_time_between_batches
  }
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

Complete logs : https://gist.github.com/beparmentier/f62db645581c1e232655aa268413d04a

```hcl
Stack trace from the terraform-provider-azurerm_v2.72.0_x5 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4977dda]
goroutine 109 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.virtualMachineScaleSetUpdateMetaData.performUpdate(0x101, 0xc0000c3140, 0xc001ca6120, 0xc0015a0708, 0x0, 0xc001d33800, 0x0, 0xc0015a1d40, 0x0, 0xc000d746c0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/virtual_machine_scale_set_update.go:38 +0x47a
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.resourceLinuxVirtualMachineScaleSetUpdate(0xc001d32980, 0x580f2c0, 0xc0009e1880, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go:842 +0xc30
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0000220e0, 0x67b0388, 0xc001b29c00, 0xc001d32980, 0x580f2c0, 0xc0009e1880, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:352 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000220e0, 0x67b0388, 0xc001b29c00, 0xc0010e9650, 0xc001775aa0, 0x580f2c0, 0xc0009e1880, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000e48438, 0x67b0388, 0xc001b29c00, 0xc001b96730, 0xc001b29c00, 0x5cf39c0, 0xc001d18600)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0018fb700, 0x67b0430, 0xc001b29c00, 0xc0010e9420, 0xc0018fb700, 0xc00194dba0, 0x46d107)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5cf39c0, 0xc0018fb700, 0x67b0430, 0xc001d18630, 0xc001b275c0, 0x0, 0x67b0430, 0xc001d18630, 0xc001d38000, 0x29d1)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00022a8c0, 0x67fd1b8, 0xc001939200, 0xc001d170e0, 0xc00115d170, 0x9d78120, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00022a8c0, 0x67fd1b8, 0xc001939200, 0xc001d170e0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00097afd0, 0xc00022a8c0, 0x67fd1b8, 0xc001939200, 0xc001d170e0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd
Error: The terraform-provider-azurerm_v2.72.0_x5 plugin crashed!
```

### Expected Behaviour

<!--- What should have happened? --->

```hcl
Terraform will perform the following actions:
  # azurerm_linux_virtual_machine_scale_set.scalesetnginx["westeurope"] will be updated in-place
  ~ resource "azurerm_linux_virtual_machine_scale_set" "scalesetnginx" {
        id                                                = "/subscriptions/[MASKED]/resourceGroups/rg_npr_int_mdm_dev_we/providers/Microsoft.Compute/virtualMachineScaleSets/vmss-npr-int-mdm-nginx-dev"
        name                                              = "vmss-npr-int-mdm-nginx-dev"
        tags                                              = {
            "bu"                = "int"
            "createdby"         = "terraform"
            "domain"            = "mdm"
            "env"               = "dev"
            "meta_environment"  = "npr"
            "platform"          = "digimdm"
            "project"           = "digimdm"
            "service"           = "[MASKED]"
            "status"            = "used"
            "subscription_name" = "sub_npr_int_digimdm"
            "team"              = "infra:sysops"
        }
        # (27 unchanged attributes hidden)
      ~ rolling_upgrade_policy {
          ~ max_batch_instance_percent              = 20 -> 100
          ~ max_unhealthy_instance_percent          = 20 -> 100
          ~ max_unhealthy_upgraded_instance_percent = 20 -> 100
          ~ pause_time_between_batches              = "PT0S" -> "PT120S"
        }
        # (6 unchanged blocks hidden)
    }
```

### Actual Behaviour

<!--- What actually happened? --->

Crash

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/internal/services/compute/linux_virtual_machine_scale_set_resource.go
+++ b/internal/services/compute/linux_virtual_machine_scale_set_resource.go
@@ -610,7 +610,7 @@ func resourceLinuxVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData, meta i
 			}
 		} else {
 			upgradePolicy = *existing.VirtualMachineScaleSetProperties.UpgradePolicy
-			existing.VirtualMachineScaleSetProperties.UpgradePolicy.Mode = compute.UpgradeMode(d.Get("upgrade_mode").(string))
+			upgradePolicy.Mode = compute.UpgradeMode(d.Get("upgrade_mode").(string))
 		}
 
 		if d.HasChange("automatic_os_upgrade_policy") {

--- a/internal/services/compute/linux_virtual_machine_scale_set_resource.go
+++ b/internal/services/compute/linux_virtual_machine_scale_set_resource.go
@@ -603,8 +603,14 @@ func resourceLinuxVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData, meta i
 	}
 
 	if d.HasChange("automatic_os_upgrade_policy") || d.HasChange("rolling_upgrade_policy") {
-		upgradePolicy := compute.UpgradePolicy{
-			Mode: compute.UpgradeMode(d.Get("upgrade_mode").(string)),
+		upgradePolicy := compute.UpgradePolicy{}
+		if existing.VirtualMachineScaleSetProperties.UpgradePolicy == nil {
+			upgradePolicy = compute.UpgradePolicy{
+				Mode: compute.UpgradeMode(d.Get("upgrade_mode").(string)),
+			}
+		} else {
+			upgradePolicy = *existing.VirtualMachineScaleSetProperties.UpgradePolicy
+			existing.VirtualMachineScaleSetProperties.UpgradePolicy.Mode = compute.UpgradeMode(d.Get("upgrade_mode").(string))
 		}
 
 		if d.HasChange("automatic_os_upgrade_policy") {

--- a/internal/services/compute/windows_virtual_machine_scale_set_resource.go
+++ b/internal/services/compute/windows_virtual_machine_scale_set_resource.go
@@ -631,8 +631,14 @@ func resourceWindowsVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData, meta
 		}
 	}
 	if d.HasChange("automatic_os_upgrade_policy") || d.HasChange("rolling_upgrade_policy") {
-		upgradePolicy := compute.UpgradePolicy{
-			Mode: upgradeMode,
+		upgradePolicy := compute.UpgradePolicy{}
+		if existing.VirtualMachineScaleSetProperties.UpgradePolicy == nil {
+			upgradePolicy = compute.UpgradePolicy{
+				Mode: compute.UpgradeMode(d.Get("upgrade_mode").(string)),
+			}
+		} else {
+			upgradePolicy = *existing.VirtualMachineScaleSetProperties.UpgradePolicy
+			existing.VirtualMachineScaleSetProperties.UpgradePolicy.Mode = compute.UpgradeMode(d.Get("upgrade_mode").(string))
 		}
 
 		if d.HasChange("automatic_os_upgrade_policy") {

--- a/internal/services/compute/windows_virtual_machine_scale_set_resource.go
+++ b/internal/services/compute/windows_virtual_machine_scale_set_resource.go
@@ -638,7 +638,7 @@ func resourceWindowsVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData, meta
 			}
 		} else {
 			upgradePolicy = *existing.VirtualMachineScaleSetProperties.UpgradePolicy
-			existing.VirtualMachineScaleSetProperties.UpgradePolicy.Mode = compute.UpgradeMode(d.Get("upgrade_mode").(string))
+			upgradePolicy.Mode = compute.UpgradeMode(d.Get("upgrade_mode").(string))
 		}
 
 		if d.HasChange("automatic_os_upgrade_policy") {

